### PR TITLE
ci: use env variable to store branch name

### PR DESCRIPTION
**Pull request by ferozsalam**
Instead of using the branch name directly in the run command.

### CODE REVIEW

File: .github/workflows/build-images-base.yaml [hunk 1]
```diff
@@ -266,10 +266,12 @@ jobs:
           APP_ID: ${{ secrets.AUTO_COMMITTER_APP_ID }}
 
       - name: Push changes into PR
+        env:
+          REF: ${{ github.event.pull_request.head.ref }}
         if: ${{ steps.cilium-runtime-tag-in-repositories.outputs.exists == 'false' || steps.cilium-builder-tag-in-repositories.outputs.exists == 'false' }}
         run: |
           git diff HEAD^
-          git push https://x-access-token:${{ steps.get_token.outputs.app_token }}@github.com/${{ env.QUAY_ORGANIZATION }}/cilium.git HEAD:${{ github.event.pull_request.head.ref }}
+          git push https://x-access-token:${{ steps.get_token.outputs.app_token }}@github.com/${{ env.QUAY_ORGANIZATION }}/cilium.git HEAD:"$REF"
```
Comment (viktor-kurchenko) [inline .github/workflows/build-images-base.yaml:274]: Do we need quotes here?

Comment (ferozsalam) [reply .github/workflows/build-images-base.yaml:274]: The quotes will prevent splitting of the variable value, which prevents command injection.

**Review by viktor-kurchenko — APPROVED** [review]
LGTM.

**Review by nbusseneau — APPROVED** [review]

**Comment by ferozsalam** [comment]
As this is a CI change I don't believe this would benefit from running the extra `/test` suite. Marking as `ready-to-merge`.

**Pull request merged**